File: tests/fixtures/patterns-positive/state/mutB.py
# POSITIVO (mutación B: State aplanado en if/elif sobre un discriminante).
class TrafficLight:
    def __init__(self):
        self.kind = "red"

    def advance(self):
        if self.kind == "red":
            self.kind = "green"
        elif self.kind == "green":
            self.kind = "yellow"
        elif self.kind == "yellow":
            self.kind = "red"

    def force_state(self, kind):
        self.kind = kind

File: tests/fixtures/patterns-positive/state/mutB.ts
// POSITIVO (mutación B: State aplanado en if/else sobre un discriminante).
class TrafficLight {
  private kind: "red" | "green" | "yellow" = "red";

  advance(): void {
    if (this.kind === "red") {
      this.kind = "green";
    } else if (this.kind === "green") {
      this.kind = "yellow";
    } else if (this.kind === "yellow") {
      this.kind = "red";
    }
  }

  forceState(kind: "red" | "green" | "yellow"): void {
    this.kind = kind;
  }
}
